testplan/testing/base.py

```diff
@@ -518,31 +519,18 @@ def _get_hook_context(self, case_report):
             case_report.logged_exceptions(),
         )
 
-    def _run_resource_hook(self, hook, label):
```

Comment:
_run_resource_hook is step that we add to runnable execution flow, please don't rename it to _prepare_suite_report, which is not as clear about what we are doing.

    def add_start_resource_steps(self):
        self._add_step(
            self._run_resource_hook,
            hook=self.cfg.before_start,
            label="Before Start",
        )

        self._add_step(self.resources.start)

        self._add_step(
            self._run_resource_hook,
            hook=self.cfg.after_start,
            label="After Start",
        )


You can add a separate error_handler step to steps, which will be skips if no error happened in previous steps.

Reply:
I haven't renamed it, it's there, please see Line 600. Github is not so intelligent when one moves sections of code...

Reply:
I only needed a few lines of codes from it and I reorganized them to another function to prevent code duplication.